### ISSUE TITLE
BindAllInterfaces<T> to bind To<T>() by default

### ISSUE THREAD

**Issue by green-coder** [ssannandeji/Zenject-2019#147]
Given this `ITickable` concrete class:

``` c#
public class PlayerManager : ITickable {
}
```

In the installer :

``` c#
    // Does not work.
    //Container.BindAllInterfaces<PlayerManager>().AsSingle().NonLazy();

    // Does not work.
    //Container.BindAllInterfaces<PlayerManager>().ToSelf().AsSingle().NonLazy();

    // This works.
    Container.BindAllInterfaces<PlayerManager>().To<PlayerManager>().AsSingle().NonLazy();
```

The error I get is Zenject complaining about `ITickable` not being a concrete type when it wants to instantiate it, so I think the default binding binds to `ITickable` type instead of `PlayerManager`.

I don't know if it is a bug or intended, but for sure it would be useful to have `BindAllInterfaces<T>` to bind `To<T>()` by default.

**Comment by green-coder** [comment]
`BindAllInterfacesAndSelf` has the same issue.

**Comment by svermeulen** [comment]
Yes, this is a very common gotcha that a lot of people experience.  I'm not sure anything should be done about it though, since technically it is following the convention used everywhere else

Given the following class:

```
public class Foo : ITickable, IInitializable {
}
```

When you write `Container.BindAllInterfaces<Foo>().AsSingle()` that gets expanded to

```
Container.Bind(typeof(IInitializable), typeof(ITickable)).ToSelf().AsSingle();
```

Which itself gets expanded to:

```
Container.Bind(typeof(IInitializable)).ToSelf().AsSingle();
Container.Bind(typeof(ITickable)).ToSelf().AsSingle();
```

Which doesn't make any sense since you can't instantiate an interface.

In other words, the default behaviour is always to assume ToSelf when unspecified.  So it would be inconsistent to change this for `BindAllInterfaces` and `BindAllInterfacesAndSelf`.

**Comment by svermeulen** [comment]
Reopening to at least add better error logging for this very common mistake and possibly a less verbose alternative

**Comment by svermeulen** [comment]
I've fixed this in version 5.0 (develop branch) by changing:

```
Container.BindAllInterfaces<Foo>().To<Foo>().AsSingle();
Container.BindAllInterfacesAndSelf<Foo>().To<Foo>().AsSingle();
```

To:

```
Container.BindInterfacesTo<Foo>().AsSingle();
Container.BindInterfacesAndSelfTo<Foo>().AsSingle();
```
